### PR TITLE
Add test for using fuzziness parameter in multi_match query

### LINKED ISSUE #18710: Using fuzziness parameter in multi_match query interferes with per field boosting
To reproduce, we start with an index with default settings containing two records in different shards and a query that boosts one of the fields:
```
DELETE test

PUT test/doc/1
{
  "title": "bar",
  "body": "foo"
}

PUT test/doc/2
{
  "title": "foo",
  "body": "bar"
}

GET test/doc/_search
{
  "query": {
    "multi_match": {
      "query": "foo",
      "fields": ["title^100", "body"]
    }
  }
}
```
We get back a reasonable result: 
```
{
  "took" : 98,
  "timed_out" : false,
  "_shards" : {
    "total" : 5,
    "successful" : 5,
    "failed" : 0
  },
  "hits" : {
    "total" : 2,
    "max_score" : 1.0,
    "hits" : [ {
      "_index" : "tests",
      "_type" : "test",
      "_id" : "2",
      "_score" : 1.0,
      "_source" : {
        "title" : "foo",
        "body" : "bar"
      }
    }, {
      "_index" : "tests",
      "_type" : "test",
      "_id" : "1",
      "_score" : 0.0059061614,
      "_source" : {
        "title" : "bar",
        "body" : "foo"
      }
    } ]
  }
}
```

Now, add the `fuzziness` parameter to the query (the value doesn't matter as long as it doesn't interfere with matches):
```
GET test/doc/_search
{
  "query": {
    "multi_match": {
      "query": "foo",
      "fields": ["title^100", "body"],
      "fuzziness": 0
    }
  }
}
```
the result is both records are scored the same:
```
{
  "took" : 17,
  "timed_out" : false,
  "_shards" : {
    "total" : 5,
    "successful" : 5,
    "failed" : 0
  },
  "hits" : {
    "total" : 2,
    "max_score" : 1.0,
    "hits" : [ {
      "_index" : "tests",
      "_type" : "test",
      "_id" : "2",
      "_score" : 1.0,
      "_source" : {
        "title" : "foo",
        "body" : "bar"
      }
    }, {
      "_index" : "tests",
      "_type" : "test",
      "_id" : "1",
      "_score" : 1.0,
      "_source" : {
        "title" : "bar",
        "body" : "foo"
      }
    } ]
  }
}
```
However, the same experiment repeated on a single shard index produces reasonable results. It looks like if the fuzziness parameter is present, queryNorm is correctly applied only on shards where the boosted field matched at least one record.

I was able to reproduce the issue on both 2.3.3 and the latest master.

The issue was originally reported on the [discussion forum](https://discuss.elastic.co/t/ne-rabotaet-boost-v-zaprosah/51648).

### PR DESCRIPTION
**Pull request by cbuescher**
There was an issue with using fuzziness parameter in multi_match query that has been reported in #18710 and was fixed in Lucene 6.2 that is now used on master. In order to verify that fix and close the original issue this PR adds the test from that issue as an integration test.

Closes #18710

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/search/query/MultiMatchQueryIT.java [hunk 1]
```diff
@@ -635,6 +635,37 @@ public void testCrossFieldMode() throws ExecutionException, InterruptedException
         assertFirstHit(searchResponse, hasId("ultimate1"));
     }
 
+    /**
+     * see https://github.com/elastic/elasticsearch/issues/18710
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/search/query/MultiMatchQueryIT.java]: I'm fine with adding a link in the javadocs but could you add a short explanation so I don't have to go read the link? When I'm scanning javadocs there is 0% chance I'll go and read the link.

File: core/src/test/java/org/elasticsearch/search/query/MultiMatchQueryIT.java [hunk 2]
```diff
@@ -635,6 +635,37 @@ public void testCrossFieldMode() throws ExecutionException, InterruptedException
...
+        assertAcked(builder.addMapping("type", "title", "type=string", "body", "type=string"));
+        ensureGreen();
+        List<IndexRequestBuilder> builders = new ArrayList<>();
+        builders.add(client().prepareIndex(idx, "type", "1").setSource(
+                "title", "foo",
+                "body", "bar"));
+        builders.add(client().prepareIndex(idx, "type", "2").setSource(
+                "title", "bar",
+                "body", "foo"));
+        indexRandom(true, false, builders);
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/search/query/MultiMatchQueryIT.java]: If it is really important that these docs end up on separate shards can you add an assertion that they are on separate shards? That would explain the `true, false, builder` thing.

**Review by nik9000 — CHANGES_REQUESTED** [review]

**Comment by cbuescher** [comment]
@nik9000 thanks for the review, I added a hopefully better descrition to the test and also added checks that the two test documents live on different shards. This should communicate better that this test relies on this setup.

**Review by nik9000 — APPROVED** [review]

**Comment by nik9000** [comment]
Thanks for making the changes! Sorry I didn't notice that you'd made them.

**Pull request merged**